Tests/Algorithm/AlgorithmLiveTradingTests.cs

```diff
@@ -48,8 +48,7 @@ public void SetHoldingsTakesIntoAccountPendingMarketOrders()
             var transactionHandler = new BrokerageTransactionHandler();
 
             transactionHandler.Initialize(algorithm, brokerage, new LiveTradingResultHandler());
-            new Thread(transactionHandler.Run) { IsBackground = true }.Start();
-            Thread.Sleep(2000);
+            Thread.Sleep(250);
```

Comment:
hmm, `Thread.Sleep(x)` always has a certain [smell](https://en.wikipedia.org/wiki/Code_smell) to me. I don't think this is important to tackle in this PR, but more for just my philosophy and how one could change it -- We could define an `WaitHandle ITransactionHandler.Initialized` and then just wait for init to complete.

Consider someone setting up a SQL database or maybe they need to fetch remote files in the initializer -- yes, esoteric examples, and simply food for throught